### DISCUSSION TITLE
[Question] Why fribidi in xmake-repo use  meson on Windows and autoconf on macOS and Linux?

### DISCUSSION THREAD

**Discussion by DreamOfIce** [xmake-io/xmake#3979] [Q&A] [answered]
Is there any reason for this? Why don't we use meson on all systems?
BTW, I am trying to add more system support for fribidi(especially wasm). Can I try to switch them all to meson?

**Comment by waruqi** [comment, marked as answer]
这个包也不是我搞的，我猜应该是最先搞了 linux, macos ，用了 autoconf。。后来为了支持 windows 所以才用了 meson

而且用 meson，还得额外安装 meson, python, ninja, openssl, zlib, libffi 等一堆依赖，编译也得半天。。相比 autoconf 。能用 autoconf 还是用 autoconf 。。无依赖安装，更快更省事。。

只有不支持的平台，可以切到 meson 去安装

**Comment by DreamOfIce** [comment]
行，但wasm似乎还是会遇到上次那个libtool的问题

**Reply by waruqi** [reply]
如果有问题，wasm 可以切 meson 试试
